server/src/main/java/org/elasticsearch/search/internal/SearchContext.java

```diff
@@ -347,6 +348,14 @@ public SearchLookup lookup() {
      */
     public abstract Profilers getProfilers();
 
+    public boolean isNested() {
```

Comment:
We shouldn't need this, we use a `SubSearchContext` for inner hits and top hits aggregator so there should be a way to set the nested path when executing the fetch phase ?